app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision/cloning.rb

```diff
@@ -4,6 +4,90 @@ def log_clone_options(clone_options)
   end
 
   def prepare_for_clone_task
+    request_type == 'clone_to_template' ? prepare_for_clone_to_template : prepare_for_clone
+  end
+
+  def start_clone(clone_options)
+    begin
+      if request_type == 'clone_to_template'
+        source.with_provider_connection(:service => "PCloudPVMInstancesApi") do |api|
+          vm = Vm.find(get_option(:src_vm_id))
+          body = IbmCloudPower::PVMInstanceCapture.new(clone_options)
+          response = api.pcloud_pvminstances_capture_post(cloud_instance_id, vm.uid_ems, body)
+          response[:taskID]
+        end
+      elsif sap_image?
+        source.with_provider_connection(:service => "PCloudSAPApi") do |api|
+          body = IbmCloudPower::SAPCreate.new(clone_options)
+          response = api.pcloud_sap_post(cloud_instance_id, body)
+          response&.first&.pvm_instance_id
+        end
+      else
+        source.with_provider_connection(:service => "PCloudPVMInstancesApi") do |api|
+          body = IbmCloudPower::PVMInstanceCreate.new(clone_options)
+          response = api.pcloud_pvminstances_post(cloud_instance_id, body)
+          response&.first&.pvm_instance_id
+        end
+      end
```

Comment:
NOTE I do think that we can clean this up by splitting into methods rather than all in this one conditional (basically what codeclimate is complaining about) but can be a follow-up